ci_framework/roles/polarion/tasks/main.yml

```diff
@@ -0,0 +1,106 @@
+
+- name: Look for test result XML files in artifacts directory
+  ansible.builtin.find:
+    paths: "{{ cifmw_polarion_jump_result_dir }}"
+    patterns: "*.xml"
+    recurse: yes
+  register: xml_files
+
+- block:
+  - name: Clone Jump repository
+    ansible.builtin.git:
+      repo: "{{ cifmw_polarion_jump_repo_url }}"
+      dest: "{{ cifmw_polarion_jump_repo_dir }}"
+      version: "{{ cifmw_polarion_jump_repo_branch }}"
+      force: true
+
+  - name: Create virtualenv
+    ansible.builtin.pip:
+      name: pip
+      virtualenv: "{{ cifmw_polarion_jump_repo_dir }}/jump-venv"
+      virtualenv_command: "/usr/bin/python3 -m venv"
+      state: latest
+      extra_args: --upgrade
+    when:
+      - cifmw_polarion_jump_repo_dir is defined and cifmw_polarion_jump_repo_dir != ""
+
+  - name: Use polarion-staging
+    ansible.builtin.replace:
+      path: "{{ cifmw_polarion_jump_repo_dir }}/{{ item }}"
+      regexp: "{{ polarion_fqdn }}"
+      replace: "{{ polarion_stage_fqdn }}"
+    loop:
+      - "helpers.py"
+      - ".pylero"
+    when:
+      - polarion_fqdn is defined and polarion_fqdn != ""
+      - polarion_stage_fqdn is defined and polarion_stage_fqdn != ""
+
+  - name: Run prepare_pylero.sh
+    ansible.builtin.shell: >
+      source jump-venv/bin/activate && \
+      ./prepare_pylero.sh
+    args:
+      chdir: "{{ cifmw_polarion_jump_repo_dir }}"
+      executable: /bin/bash
+    register: pylero_output
+
+  - name: Output prepare_pylero
+    ansible.builtin.debug:
+      var: pylero_output.stdout_lines
+
+  - name: Validate XML files with xmllint
+    ansible.builtin.command: "xmllint --noout {{ item.path }}"
+    loop: "{{ xml_files.files }}"
+    register: xmllint_output
+    changed_when: false
+    failed_when: xmllint_output.rc != 0
+
+  - name: Set XML files in a fact
+    ansible.builtin.set_fact:
+      found_xml_files: "{{ xml_files.files | map(attribute='path') | join(',') }}"
+
+  - name: Update Test Runs
+    ansible.builtin.shell: >
+      source "{{ cifmw_polarion_jump_repo_dir }}/jump-venv/bin/activate" && \
+      python jump.py \
+        --testrun-id={{ cifmw_polarion_test_id }} \
+        --xml-file={{ found_xml_files }} \
+        --update_testcases={{ cifmw_polarion_update_testcases | default(false) }}
+  #       --dfg Network
+  #       --jenkins_build_url="{{ cifmw_polarion_zuul_build }}" \
+  #       --puddle-id="{{ cifmw_polarion_osp_puddle_id }}" \
+  #       --custom-fields build="{{ cifmw_polarion_osp_release }}" \
+  #       --remove-old-tests="{{ cifmw_polarion_remove_old_test_cases_itr }}" \
+  #       --update-existing-test-cases="{{ cifmw_polarion_update_existing_test_cases }}"
+    register: jump_script
+    args:
+      chdir: "{{ cifmw_polarion_jump_repo_dir }}"
+
+  - name: Output of jump
+    ansible.builtin.debug:
+      var: jump_script.stderr_lines
+
+  when: xml_files.files | length > 0
```

Comment:
care to push that `when:` closer to the related `name` ? According to the indents, it should be somewhere near Line 31? (see - it's not really friendly ;))

Reply:
Absolutely, moved.